src/connector.ts

```diff
@@ -52,29 +55,40 @@ export declare interface DriverOptions {
 // adds extra logic to async initialize items.
 class CloudSQLInstanceMap extends Map {
   async loadInstance({
-    type,
+    ipType,
+    authType,
     instanceConnectionName,
     sqlAdminFetcher,
   }: {
-    type: IpAdressesTypes;
+    ipType: IpAdressesTypes;
+    authType: AuthTypes;
     instanceConnectionName: string;
     sqlAdminFetcher: SQLAdminFetcher;
   }): Promise<void> {
     // in case an instance to that connection name has already
     // been setup there's no need to set it up again
-    if (this.has(instanceConnectionName)) {
+    if (this.has(this.getKey({instanceConnectionName, ipType, authType}))) {
       return;
     }
     const connectionInstance = await CloudSQLInstance.getCloudSQLInstance({
-      connectionType: type,
+      connectionType: ipType,
+      authType: authType,
```

Comment:
when the key name and the assigning value var name match, you can simplify the declaration by just declaring that name once:

```suggestion
      authType,
```

Reply:
fixed!